rename promtool based gh worklow jobs

bavarianbidi opened the pull request:
This PR renamed the `promtool test` based unit tests from `test/gitleaks` --> `promtool-tests/promtool-unit-tests`.
It also fixes a broken unit test which has landed on `master` due non-blocking permissions on `master` branch.

### Checklist

- [x] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

hervenicol commented:
100% agree the name was bad.
However, these are not only promtool tests, we have added for instance inhibition labels tests (https://github.com/giantswarm/prometheus-rules#test-inhibition-labels).

hervenicol commented on CHANGELOG.md:
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
+### Changed
+
+- rename github worklow based unit tests
```
```suggestion
- rename github workflow based unit tests
```

bavarianbidi commented:
> 100% agree the `gitleaks` name was very bad. However, these are not only promtool tests, we have added for instance inhibition labels tests ([giantswarm/prometheus-rules#test-inhibition-labels](https://github.com/giantswarm/prometheus-rules#test-inhibition-labels)).

thx for the info, i didn't checked/know that new tests where added.
For that i renamed the github worklows to make it more clear

![image](https://user-images.githubusercontent.com/21329055/217063279-1b1d6441-39b8-41a7-9fd5-df11149dbfd3.png)
